### PR TITLE
Refactor _OS, os.get() & os.is()

### PR DESCRIPTION
**Pull request by tvandijck**
This fixes: https://github.com/premake/premake-core/issues/722

replaced:
```
os.get() -> os.target()
os.is() -> os.istarget()
_OS -> _TARGET_OS
action.os -> action.targetos
```

added:
```
os.host()
os.ishost()
```

### CODE REVIEW

File: src/host/os_current.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * \file   os_current.c
+ * \brief  Get the current OS we're executing on.
+ * \author Copyright (c) 2014-2017 Tom van Dijck, Jason Perkins and the Premake project
+ */
+
+#include "premake.h"
+
+int os_current(lua_State* L)
```
Comment (tvandijck) [inline src/host/os_current.c]: I did it this way, because I want it to be absolutely read-only at runtime.

**Comment by tritao** [comment]
This is a much needed clarification! 👍 

I wonder if we instead of "current" we should use "host" OS term instead, but might be just be my bias showing due to dealing a lot with that terminology with cross compilers.

**Comment by tvandijck** [comment]
I like ```host``` it does seem a little more explicit..

File: src/base/os.lua [hunk 2]
```diff
@@ -134,26 +134,65 @@
...
+		__newindex = function(t, k, v)
+			if (k == '_OS') then
+				premake.warnOnce("_OS+set", "_OS is deprecated, use '_TARGET_OS'.")
+				rawset(t, "_TARGET_OS", v)
+			else
+				rawset(t, k, v)
+			end
+		end
+	}
+	setmetatable(_G, _G_metatable)
```
Comment (samsinsane) [inline src/base/os.lua]: Perhaps this should be a helper for variables that are deprecated in the future?

Comment (tvandijck) [inline src/base/os.lua]: This method actually has quite the impact on performance... ANY access to a global, now has a string compare first.. so I wanted to keep this as simple as possible, and really only keep it in there for the next 2-3 alpha releases. I really don't want to write a 'generic' deprecated globals table...

**Pull request merged**